source/Mlos.Python/mlos/Optimizers/BayesianOptimizer.py

```diff
@@ -59,20 +61,47 @@ def __init__(
 
         # Now let's put together the surrogate model.
         #
+        print(f'self.optimizer_config.surrogate_model_implementation: {self.optimizer_config.surrogate_model_implementation}')
         assert self.optimizer_config.surrogate_model_implementation in (
             HomogeneousRandomForestRegressionModel.__name__,
-            MultiObjectiveHomogeneousRandomForest.__name__
+            MultiObjectiveHomogeneousRandomForest.__name__,
+            MultiObjectiveLassoCrossValidated.__name__,
+            MultiObjectiveRegressionEnhancedRandomForest.__name__
         )
 
         # Note that even if the user requested a HomogeneousRandomForestRegressionModel, we still create a MultiObjectiveRegressionModel
         # with just a single RandomForest inside it. This means we have to maintain only a single interface.
         #
-        self.surrogate_model: MultiObjectiveRegressionModel = MultiObjectiveHomogeneousRandomForest(
-            model_config=self.optimizer_config.homogeneous_random_forest_regression_model_config,
-            input_space=self.optimization_problem.feature_space,
-            output_space=self.surrogate_model_output_space,
-            logger=self.logger
-        )
+        if self.optimizer_config.surrogate_model_implementation == HomogeneousRandomForestRegressionModel.__name__:

```

Comment:
This is great! Now that we have so many regression models, I'll move this code to a RegressionModelFactory - it should make testing much easier :)